### PR TITLE
Made CustomEntityBundle (reference) docs official

### PR DESCRIPTION
**Pull request by mickaelandrieu**
The CustomEntityBundle is a "MUST-GO" when you want to provide an UI for your reference data.

As this bundle will become in a mid-term part of Akeneo, it's time to move docs of Bundle and reword it a little bit.

update: this pull request is part of a bigger plan to totally document what is a reference data, see https://github.com/akeneo/pim-docs/issues/488

> Needed to be cherry picked in ``1.7``, ``1.6`` and ``1.5`` branches.

/!\ don't merge it => should be done in ``1.5`` branch

### CODE REVIEW

**Comment by mickaelandrieu** [comment]
@jmleroux I disagree, this is the documentation of a bundle that "ease" reference data, the reference data are provided by the core and should be documented too.

This is not the scope of this pull request, that document only what is provided by the bundle.

I don't know if I'm clear, or if I have well understood what is your request :D

File: reference/bundles/custom_entity_bundle/index.rst [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+The Custom Entity Bundle facilitates the creation of PIM reference data and related views in Akeneo.
+
+Requirements
+************
+
+The bundle is not synchronized with the Akeneo development release cycle, look at this table
+to choose the right version to install:
+
++-------------------------+-----------------------------------+
+| **CustomEntitBundle**   | **Akeneo PIM Community Edition**  |
```
Comment (nidup) [inline reference/bundles/custom_entity_bundle/index.rst]: CustomEntityBundle

Comment (mickaelandrieu) [inline reference/bundles/custom_entity_bundle/index.rst]: Fixed, thanks

**Comment by jmleroux** [comment]
> @jmleroux I disagree, this is the documentation of a bundle that "ease" reference data, the reference data are provided by the core and should be documented too.

OK, `reference data` should be documented, and maybe not here. You take his point.
So let's answer to another question: what is a custom entity? I really do think that a little concrete example of what is a custom entity would be usefull. This bundle is about adding easy CRUD for reference data, ie adding concrete implementation for the very technical `reference data`.

**Comment by mickaelandrieu** [comment]
@jmleroux I will add a concrete implementation in a new cookbook called "create a crud for a reference data" which is the following of the one about creation, I wanted to but I needed to explain so much in only one cookbook that I needed first to create docs like this one - more abstract, but also more complete - before.

It's always hard to see the difference between reference and cookbook. To me, a reference is complete and describes patterns and all options when a cookbook is more like a recipe.

We need both here: a reference documentation for reference data, a cookbook for crud creation and ... a reference for the bundle needed.

It's my fault, I've didn't explained this contribution is just a "start" and will be followed by - at least - 2 others, covering your needs, in some parts :)

What do you think ?

**Comment by jmleroux** [comment]
> What do you think ?

All i want is clear documentation about what is a reference data / custom entity.
If it is in your backlog, i cannot do anything else but applause :)

**Comment by mickaelandrieu** [comment]
closed in favor of https://github.com/akeneo/pim-docs/pull/495

**Pull request closed without merging**